pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -441,6 +441,9 @@ func (p *PriorityQueue) activate(pod *v1.Pod) bool {
 // isPodBackingoff returns true if a pod is still waiting for its backoff timer.
 // If this returns true, the pod should not be re-tried.
 func (p *PriorityQueue) isPodBackingoff(podInfo *framework.QueuedPodInfo) bool {
+	if podInfo.Gated {
+		return false
```

Comment:
This is good to have for sure, but I assume that podInfo for a gated pod should never have its Timestamp set beyond the first time we observed the pod (on Add(...)) because we should have never attempted to schedule the pod (i.e., Attempts should always be zero), and so Timestamp shouldn't be reset.

Reply:
Yes, attempts is always zero, so `duration := p.calculateBackoffDuration(podInfo)` would always return `podInitialBackoffDuration`. In other words, if the duration between the pod is added and updated less than podInitialBackoffDuration, isPodBackingOff would return true.

This bug can be observed by running `hack/local-up-cluster.sh` with the following diff:

```diff
diff --git a/hack/local-up-cluster.sh b/hack/local-up-cluster.sh
index 6ec7f1ded28..e2c67dd0e34 100755
--- a/hack/local-up-cluster.sh
+++ b/hack/local-up-cluster.sh
@@ -869,6 +869,8 @@ clientConnection:
   kubeconfig: ${CERT_DIR}/scheduler.kubeconfig
 leaderElection:
   leaderElect: false
+podInitialBackoffSeconds: 120
+podMaxBackoffSeconds: 200
 EOF
     ${CONTROLPLANE_SUDO} "${GO_OUT}/kube-scheduler" \
       --v="${LOG_LEVEL}" \
```